### ISSUE TITLE
how does this app connect to the camera remotely?

### ISSUE THREAD

**Issue by alur222** [brusdev/V380-SDK#2]
hi, I know it connects via rstp

I am wondering if it uses the public IP of the router ? If so, can you give me an example rstp url string?

I was wondering why I can't connect to my camera while at the office using VLC.

Using the v380 app, I can connect without problems.

Thank you for the help!

**Comment by Gowresh7** [comment]
To view it in VLC remotely, you have to do the following:-

    Connect V380 to internet (Plug Ethernet from Router to V380)
    Find the IP Address assigned to V380 by the router.(It is advisable to reserve the IP address in router settings)
    Enable Port Forwarding in your Router for the V380's IP Address.
    Make note of your Public IP (Just Google it.)
    The V380 rtsp url is generally in the format "rtsp:///live/ch00_1". After port forwarding is enabled, the url will change to "rtsp://public-ip:port/live/ch00_1"
    Using this IP, you can live stream in VLC.
    Note: You will not be able to ptz control the camera movements. I have a different method for that purpose. Will upload the source code for that shortly.

**Comment by ketul93** [comment]
> To view it in VLC remotely, you have to do the following:-
> 
> ```
> Connect V380 to internet (Plug Ethernet from Router to V380)
> Find the IP Address assigned to V380 by the router.(It is advisable to reserve the IP address in router settings)
> Enable Port Forwarding in your Router for the V380's IP Address.
> Make note of your Public IP (Just Google it.)
> The V380 rtsp url is generally in the format "rtsp:///live/ch00_1". After port forwarding is enabled, the url will change to "rtsp://public-ip:port/live/ch00_1"
> Using this IP, you can live stream in VLC.
> Note: You will not be able to ptz control the camera movements. I have a different method for that purpose. Will upload the source code for that shortly.
> ```

How to connect for the local network using RTSP protocol in VLC?

**Comment by Gowresh7** [comment]
To connect local network, that is with the cam's network, remove the ethernet cable. The cam will start its own wireless hotspot. Connect to that network and use the above mentioned url with ip as 192.168.1.1.

To connect to local network within the ethernet connection, just use the url with cam's ip address as shown in the router.

**Comment by shreyasnagare** [comment]
> To view it in VLC remotely, you have to do the following:-
> 
> ```
> Connect V380 to internet (Plug Ethernet from Router to V380)
> Find the IP Address assigned to V380 by the router.(It is advisable to reserve the IP address in router settings)
> Enable Port Forwarding in your Router for the V380's IP Address.
> Make note of your Public IP (Just Google it.)
> The V380 rtsp url is generally in the format "rtsp:///live/ch00_1". After port forwarding is enabled, the url will change to "rtsp://public-ip:port/live/ch00_1"
> Using this IP, you can live stream in VLC.
> Note: You will not be able to ptz control the camera movements. I have a different method for that purpose. Will upload the source code for that shortly.
> ```

Keep up the good work!

Assuming I have the camera at 192.168.1.100

`rtsp://192.168.1.100/live/ch00_1`

I tried putting my login details in the network URL (for VLC) and used this instead:

`rtsp://username:password@192.168.1.100/live/ch00_1` but it didn't work either (yes, I tried it on a laptop connected to the same router)

**Comment by Gowresh7** [comment]
@shreyasnagare Did you configure the username and password in the root folder of the camera?

**Comment by shreyasnagare** [comment]
> @shreyasnagare Did you configure the username and password in the root folder of the camera?

I don't understand. If you're asking if `rtsp://username:password@192.168.1.100/` works, it doesn't.

**Comment by Gowresh7** [comment]
> > @shreyasnagare Did you configure the username and password in the root folder of the camera?
> 
> I don't understand. If you're asking if `rtsp://username:password@192.168.1.100/` works, it doesn't.

1. Are you using the default username and password ?
2. What happens when you try without the username & password?
3. Did you by any chance update the firmware of the Camera?

**Comment by shreyasnagare** [comment]
> 1. Are you using the default username and password ?

No, I'm not using the default password. I've tried putting in the default password as well tho.
> 2. What happens when you try without the username & password?

Same error.
```
Connection failed:
VLC could not connect to "192.168.1.100"
```

> 3. Did you by any chance update the firmware of the Camera?

No, I did not.

**Comment by drtanzil** [comment]
> > 1. Are you using the default username and password ?
> 
> No, I'm not using the default password. I've tried putting in the default password as well tho.
> 
> > 1. What happens when you try without the username & password?
> 
> Same error.
> 
> ```
> Connection failed:
> VLC could not connect to "192.168.1.100"
> ```
> 
> > 1. Did you by any chance update the firmware of the Camera?
> 
> No, I did not.

You need to enable rtsp first by  patching the camera firmware. Here is the ceshi.ini file provided by v380 support team to me.

step 1: copy the “ceshi” file and paste it to SD card root directory(means if there are any folders in the card, do not move them in any of them) 

step 2: insert the card into camera again and restart the camera, wait for the camera proceed automatically， it may speak some Chinese

step 3（necessary steps）: delete the file i sent you in the SD card and restart the camera

Camera needs to be powered off when the SD card is taken out or inserted in

Then you'll be able to connect rtsp:// using VLC, iSpy & others rtsp software
[ceshi.zip](https://github.com/brusdev/V380-SDK/files/5604398/ceshi.zip)

**Comment by drtanzil** [comment]
@Gowresh7 : Can you please send the method to PZT for rtsp ??

**Comment by Gowresh7** [comment]
> @Gowresh7 : Can you please send the method to PZT for rtsp ??

@drtanzil Please follow my github repository for ptz control.

[https://github.com/Gowresh7/V380_Python.git](url)

**Comment by shreyasnagare** [comment]
Thanks @drtanzil. Worked like a charm.

**Comment by sharun-s** [comment]
Anyone know how to turn off the night mode? After 6-630pm the rtsp url doesn't change but the stream comes from a different camera with the IR LEDs on. I am looking for a way to stop that switch over and keep the original stream on...

**Comment by sharun-s** [comment]
Just an update on handling night mode - turns out there is only one camera (I thought there were multiple cams to produce the panorama view - the single cam apparently does it with a fisheye/wide angle lens). 

In the evening it switches (based on light sensor?) from color to black and white mode and turns on 4 Infrared LEDs to deal with darkness. In my case this was producing an unusable feed because the camera was placed behind a glass window and the glare from the IR LEDs was messing up what was getting captured. So I disconnected the IR LEDs. The switch to black and white still happens but the area has enough light for the image to be clear.

**Comment by lwvmobile** [comment]
Anybody know how to connect to the audio stream, or if there is a separate audio stream. In the V380PRO android app, I can tap the icon to get audio playback, but I'm not sure how this is accomplished. 

As far as I can see, and have found on forums, etc. is that on these V380 variants (I have three that all look alike, but are all slightly different in features) is that you get two RTSP streams for video.

rtsp://ipaddress/live/ch00_1 gives the HD 1080p (HD by technicality, not by real image quality)
rtsp://ipaddress without anything else gives out a 640x480 mashed VGA

But I haven't figured out how audio is processed. Didn't know if it was something simple, or more complex. I currently get no audio streams available on either feed when playing back through VLC or any other

**Comment by nvbach91** [comment]
> Thanks @drtanzil. Worked like a charm.

@shreyasnagare What is the final rtsp URL you used? Did you have to configure the port forwarding?

**Comment by hamx71** [comment]
Así funciona amigo :
rtsp://IP sin puerto /live/ch00_0  sd
rtsp://IP sin puerto /live/ch00_1  hd
Así funciona. Lo malo es que no tiene autenticación .

**Comment by TorrCod** [comment]
> Anybody know how to connect to the audio stream, or if there is a separate audio stream. In the V380PRO android app, I can tap the icon to get audio playback, but I'm not sure how this is accomplished.
> 
> As far as I can see, and have found on forums, etc. is that on these V380 variants (I have three that all look alike, but are all slightly different in features) is that you get two RTSP streams for video.
> 
> rtsp://ipaddress/live/ch00_1 gives the HD 1080p (HD by technicality, not by real image quality) rtsp://ipaddress without anything else gives out a 640x480 mashed VGA
> 
> But I haven't figured out how audio is processed. Didn't know if it was something simple, or more complex. I currently get no audio streams available on either feed when playing back through VLC or any other

Did you found the solution?.. I'm having the same problems

**Comment by ingGhost** [comment]
i 

> > > 1. Are you using the default username and password ?
> > 
> > 
> > No, I'm not using the default password. I've tried putting in the default password as well tho.
> > > 1. What happens when you try without the username & password?
> > 
> > 
> > Same error.
> > ```
> > Connection failed:
> > VLC could not connect to "192.168.1.100"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > 1. Did you by any chance update the firmware of the Camera?
> > 
> > 
> > No, I did not.
> 
> You need to enable rtsp first by patching the camera firmware. Here is the ceshi.ini file provided by v380 support team to me.
> 
> step 1: copy the “ceshi” file and paste it to SD card root directory(means if there are any folders in the card, do not move them in any of them)
> 
> step 2: insert the card into camera again and restart the camera, wait for the camera proceed automatically， it may speak some Chinese
> 
> step 3（necessary steps）: delete the file i sent you in the SD card and restart the camera
> 
> Camera needs to be powered off when the SD card is taken out or inserted in
> 
> Then you'll be able to connect rtsp:// using VLC, iSpy & others rtsp software [ceshi.zip](https://github.com/brusdev/V380-SDK/files/5604398/ceshi.zip)

hii, i try this with v380 my camera. but still nor work, any solution pls? thanks for u
